### ISSUE TITLE
cypress/no-chained-get: Disallow chaining .get() after any other query

### ISSUE THREAD

**Issue by MgmClientGuy0** [cypress-io/eslint-plugin-cypress#290]
According to the documentation, the rule 'cypress/no-chained-get' has the intention of preventing unnecessary queries as .get() always starts from the root element. It correctly identified cy.get().get() chains as problematic, but chaining .get() after any other Cypress query should be unnecessary, Consider this example:

```js
cy.get("selector").find("childSelector").get("anotherSelector")
```
It ignores the DOM elements of the first queries, since it starts from the root. This is true for all kinds of queries that yield DOM elements, so closest, contains, document, eq, filter, find, first, focused, get, last, next, nextAll, nextUntil, not, parent, parents, parentsUntil, prev, prevAll, prevUntil, root, shadow, siblings.

The only problem I see is if assertions are involved. I believe the following example is technically correct, since the first query was used to assert something, although it should arguably be discouraged since the second query ignores the first subject and can be their own chain. I am fine with the rule not reporting this pattern.

```js
cy.get("selector").should("not.exist").get("anotherSelector")
```
We had several cases like this in our tests. Many developers new to Cypress did not know the difference between .get() and .find() so I believe preventing this has value.

**Comment by MikeMcC399** [comment]
In the tests for the rule [no-chained-get.md](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-chained-get.md), the code:

```js
    { code: 'cy.get(\'.div\').find().get()' },
```

is listed as `valid`

https://github.com/cypress-io/eslint-plugin-cypress/blob/f91d3a990354817e34a2658ac758dd8c1902099d/tests/lib/rules/no-chained-get.js#L18-L31

This is similar to the example:

```js
cy.get("selector").find("childSelector").get("anotherSelector")
```

Maybe @duranbe, who was the author of the rule through PR https://github.com/cypress-io/eslint-plugin-cypress/pull/249, would like to comment why `cy.get().find().get()` was considered `valid` when the rule [no-chained-get.md](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-chained-get.md) was created? Also the question would need to be answered if this usage should now be considered `invalid` and if a (breaking) rule change should therefore be made?

We may also need to ask the opinion of the Cypress.io team who approved the original PR.

## Reference

- Cypress API [.get()](https://docs.cypress.io/api/commands/get)
- [Get vs Find](https://docs.cypress.io/api/commands/get#Get-vs-Find)
- [cypress/no-chained-get](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-chained-get.md) rule

**Comment by duranbe** [comment]
Probably a miss from my side, I'd be in favor of doing a breaking change to be fair
